### DISCUSSION TITLE
How to query Waybar for its visibility state from a shell script? And how to ask Waybar to specifically show or hide itself (via a separate command for each, i.e. not "toggle")?

### DISCUSSION THREAD

**Discussion by SpaceshipOperations** [Alexays/Waybar#2098] [Q&A]
As the title says, I want to be able to know whether Waybar is currently visible or not from a shell script. Without relying on some WM tool like `swaymsg` to answer the question.

I also want to be able to ask Waybar to specifically show or hide itself, i.e. via separate "show" and "hide" commands, which means that the usual `SIGUSR1` won't do it for me because it's a **toggle** command. Again, I want to be able to communicate directly with Waybar, not relying on some WM proxy command like `swaymsg`.

Thank you in advance.

**Comment by Jedsek** [comment]
@SpaceshipOperations I found `killall -9 SIGUSR2 waybar` could `hide` it

**Comment by ghost** [comment]
> I also want to be able to ask Waybar to specifically show or hide itself, i.e. via separate "show" and "hide" commands, which means that the usual `SIGUSR1` won't do it for me because it's a **toggle** command

Any progress on this ??
